### PR TITLE
feat(error-page): add error page component

### LINKED ISSUE #188: Browsing to a non-existing webpage should display a 5xx error page
## Expected Behavior
Browsing to a non-existing address should display a 5xx error page.

## Current Behavior
It shows a blank page (loads nothing)

## Possible Solution
Update the DAppContainer component to listen for webview [did-fail-load](https://electronjs.org/docs/api/webview-tag#event-did-fail-load) events.  In such a case, render a custom component that gives details about the `errorCode` and `errorDescription`.

reference: https://github.com/nos/client/issues/188#issuecomment-395820523

## Steps to Reproduce (for bugs)
1. Surf to any website/dapp (either on the nOS OR http(s) protocol) that does not exist
2. Look

## Context
It's not clear that it doesn't exist

## Your Environment
any

### PR DESCRIPTION
**Pull request by robmoorman**
## Description
This PR implements the business logic required to show any type of error when a page could not be loaded (implementation of specific errors could be done in other PR's).

Take notice of Electron's errors https://cs.chromium.org/chromium/src/net/base/net_error_list.h

## Motivation and Context
Currently when, particularly, nos:// domains couldn't be loaded it's unknown what's happening. The browsers shows an empty screen. To give the user some more insights about the error we would like to provide a custom `Error` component with more details (which target, error and description of the error).

## How Has This Been Tested?
- [x] Unhappy page loading flows (no internet, proxy, invalid url, etc.)
- [x] Switch between tabs, reloading pages and changing urls after errors
- [x] Enhanced the testsuite with an error tab (see `tab-3`)
- [x] nOS Local / localhost / dApp startkit

## Screenshots (if appropriate):
screen shot 2018-06-12 at 23 04 50

## Types of changes
- [x] Chore (tests, refactors, and fixes)
- [x] New feature (adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:
- [x] My code follows the code style of this project.
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] I have read the **CONTRIBUTING** document.

## Closing issues
Fixes #188

### CODE REVIEW

File: src/browser/components/Error/index.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import { connect } from 'react-redux';
+
+import Error from './Error';
+
+export default connect(null, null)(Error);
```
Comment (robmoorman) [inline src/browser/components/Error/index.js]: Connect in place, so we can quickly add actions here (e.g. reload target)

File: src/browser/components/Error/Error.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  static propTypes = {
+    target: string.isRequired,
+    code: number.isRequired,
+    description: string.isRequired
+  }
+
+  render() {
+    return (
+      <div className={styles.error}>
+        {this.renderTitle()}
```
Comment (robmoorman) [inline src/browser/components/Error/Error.js]: Custom implementation of specific error handling could be done here

File: src/browser/components/DAppContainer/DAppContainer.js [hunk 3]
```diff
@@ -84,6 +87,36 @@ export default class DAppContainer extends React.Component {
     );
   }
 
+  renderError() {
+    const { target, errorCode, errorDescription } = this.props;
+
+    // Validate if there is an error code (do not check with ! as the code
+    // could have a negative value)
+    if (errorCode !== null) {
```
Comment (mhuggins) [inline src/browser/components/DAppContainer/DAppContainer.js]: Very minor, but can we inverse this to make it a guard clause?  i.e.:

```js
if (errorCode === null) {
  return null;
}

return (
  <Error ... />
);
```

**Comment by robmoorman** [comment]
Fixed @mhuggins thanks for the quick review and testing!

**Comment by mhuggins** [comment]
Thanks @robmoorman!

**Pull request merged**